src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8231,7 +8231,7 @@ private static bool IsExpectedAssignmentOperator(SyntaxKind kind)
 
         private bool IsPossibleAwaitExpressionStatement()
         {
-            return (this.IsInteractive || this.IsInAsync) && this.CurrentToken.ContextualKind == SyntaxKind.AwaitKeyword;
+            return (this.IsScript || this.IsInAsync) && this.CurrentToken.ContextualKind == SyntaxKind.AwaitKeyword;
```

Comment:
If ```IsInteractive == true``` then ```IsScript == true``` since ```IsScript``` is defined as ```Options.Kind != SourceCodeKind.Regular```. The name ```IsScript``` is confusing. I'll rename it ```IsScriptOrInteractive```.